examples/favorites_advanced/bloc_favorites_advanced/lib/feature_puppy/search/blocs/puppy_list_bloc.dart

```diff
@@ -0,0 +1,91 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:favorites_advanced_base/models.dart';
+import 'package:favorites_advanced_base/repositories.dart';
+import 'package:meta/meta.dart';
+
+part 'puppy_list_event.dart';
+
+part 'puppy_list_state.dart';
+
+// part 'puppy_list_bloc_models.dart';
+
+class PuppyListBloc extends Bloc<PuppyListEvent, PuppyListState> {
+  PuppyListBloc(this.repository)
+      : super(const PuppyListState(status: PuppyListStatus.loading));
+
+  PuppiesRepository repository;
+  List<Puppy> puppyItems = [];
+
+  Future<List<Puppy>> _fetchPuppies(PuppiesRepository repository) async {
+    var puppies = <Puppy>[];
+    puppies = await repository.getPuppies(query: '');
+    if (puppies.isEmpty) {
+      throw Exception('error fetching puppies');
```

Comment:
@StanislavPrime, if there are no records in the API, does this mean that something goes wrong ?